core/src/main/scala/pureconfig/package.scala

```diff
@@ -32,15 +32,15 @@ package object pureconfig {
     }
 
     // we're not expecting any exception here, this `try` is just for extra safety
-    try getValue(ConfigCursor(conf.root(), Nil), splitPath(namespace)) catch {
+    try getValue(ConfigCursor(confValue, Nil), splitPath(namespace)) catch {
       case ex: ConfigException => ConfigReader.Result.fail(ThrowableFailure(ex, ConfigValueLocation(ex.origin())))
     }
   }
 
   // loads a value from a config in a given namespace. All `loadConfig` methods _must_ use this method to get correct
   // namespace handling, both in the values to load and in the error messages.
-  private[this] def loadValue[A](conf: TypesafeConfig, namespace: String)(implicit reader: Derivation[ConfigReader[A]]): ConfigReader.Result[A] = {
-    getValue(conf, namespace, reader.value.isInstanceOf[ReadsMissingKeys]).right.flatMap(reader.value.from)
+  private[pureconfig] def loadValue[A](configValue: ConfigValue, namespace: String)(implicit reader: Derivation[ConfigReader[A]]): ConfigReader.Result[A] = {
```

Comment:
There may be some small complications with you accepting a non-object here. What would happen if you tried `loadYaml[Option[Int]](pathToYaml, "foo")`, where `pathToYaml` points to a YAML file whose root value is not an object (e.g. an array)?

(regardless of the result, can you add a test for this corner case?)

Reply:
@ruippeixotog You're right. This would fail with `WrongType(LIST, Set(OBJECT))`. Since `getValue` does `asObjectCursor`, the method itself should only accept a `Config` and the `loadYaml` overloads should make sure the passed YAML is of an `Object` like structure, otherwise fail with with `WrongType(LIST, Set(OBJECT))`. WDYT?